app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddBarrierOpening.kt

```diff
@@ -0,0 +1,72 @@
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.WHEELCHAIR
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+
+class AddBarrierOpening(
+        private val checkArSupport: ArSupportChecker
+    ) : OsmElementQuestType<WidthAnswer> {
+
+    private val nodeFilter by lazy { """
+       nodes with
+         barrier ~ ${GATEWAYS.joinToString("|")}
+         and (!maxwidth:physical or source:maxwidth_physical ~ ".*estimat.*")
+         and (!width or source:width ~ ".*estimat.*")
+         and (!maxwidth or source:maxwidth ~ ".*estimat.*")
+         and access !~ private|no|customers|agricultural
+    """.toElementFilterExpression() }
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          highway and access ~ private|no|customers|agricultural
+    """.toElementFilterExpression() }
+
+    override val changesetComment = "Specify width of opening"
+    override val wikiLink = "Key:barrier"
+    override val icon = R.drawable.ic_quest_barrier_opening
+    override val achievements = listOf(BICYCLIST, WHEELCHAIR)
+    override val defaultDisabledMessage: Int
+        get() = if (!checkArSupport()) R.string.default_disabled_msg_no_ar else 0
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_barrier_opening_width_gate
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+        mapData.nodes.filter { nodeFilter.matches(it) }
+        val excludedWayNodeIds = mapData.ways
+            .filter { excludedWaysFilter.matches(it) }
+            .flatMapTo(HashSet()) { it.nodeIds }
+
+        return mapData.nodes
+            .filter { nodeFilter.matches(it) && it.id !in excludedWayNodeIds }
+    }
+
+
+    override fun isApplicableTo(element: Element) =
+        nodeFilter.matches(element)
+
+    override fun createForm() = AddGateWidthForm()
+
+    override fun applyAnswerTo(answer: WidthAnswer, tags: Tags, geometry: ElementGeometry, timestampEdited: Long) {
+
+        val key = if (tags["barrier"] in GATEWAYS) "maxwidth:physical" else 0
+            tags[key.toString()] = answer.width.toOsmValue()
+
+            if (answer.isARMeasurement) {
+            tags["source:$key"] = "ARCore"
+            } else {
+            tags.remove("source:$key")
+            }
+
```

Comment:
1. indentation
2. `else 0`?! What?
3. Why `maxwidth:physical` and not `maxwidth`?

Reply:
Sorry, if I try to answer these remaining problems, I will just be wasting more of your time. I've tried following the provided guide on producing a quest, and have put together sections of 3 or 4 other guides, but have reached the point where I don't have the skills to go any further. 

Sorry if I've wasted your time.

Reply:
@Nick-Tallguy if you don't feel like working on it anymore, that is of course your choice and OK; I just wanted to say your effort is appreciated, and that **_multiple rounds of asking for improvements on PR are normal and expected_**, and not intended to be taken as a discouragement, but instead as opportunity to grow and produce quality code you'd later be proud of (and which would be easy for everyone else to understand, use, and improve). 

I know it can feel overwhelming and seemingly neverending stream of improvement requests,  but (having done it several times myself) I can assure you those eventually get resolved and results in great code and no more changes are requested. (With time, one gets better and can shave off few iterations, but even for people doing it for years it is rare  for anything but must basic changes not to get that _"changes requested"_ dialogs. )

So, if you're willing, I'd recommend trying some more for those finals pushes, as you seem close to the end of the tunnel. :heart: Remember: the harder the effort, the more valued the result will be too.  

And if you get stuck, don't be afraid to ask questions and clarifications. I'm not that good at Kotlin itself, but perhaps can help with OSM and some StreetComplete logic; and there are others lurking too.

Reply:
@westnordost - I think I've updated this particular problem now